front/components/navigation/Navigation.tsx

```diff
@@ -0,0 +1,122 @@
+import { XMarkIcon } from "@dust-tt/sparkle";
+import type { SubscriptionType, WorkspaceType } from "@dust-tt/types";
+import { Dialog, Transition } from "@headlessui/react";
+import { Bars3Icon } from "@heroicons/react/20/solid";
+import { Fragment, useContext } from "react";
+
+import type { SidebarNavigation } from "@app/components/navigation/config";
+import { NavigationSidebar } from "@app/components/navigation/NavigationSidebar";
+import { SidebarContext } from "@app/components/sparkle/AppLayout";
+
+interface NavigationProps {
+  hideSidebar: boolean;
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  navChildren?: React.ReactNode;
+  subNavigation?: SidebarNavigation[] | null;
+}
+
+export function Navigation({
+  hideSidebar,
+  owner,
+  subscription,
+  navChildren,
+  subNavigation,
+}: NavigationProps) {
+  const { sidebarOpen, setSidebarOpen } = useContext(SidebarContext);
+
+  if (hideSidebar) {
+    return null;
+  }
+
+  return (
+    <div className="flex shrink-0 overflow-x-hidden">
+      {/* Mobile sidebar */}
+      <div className="fixed left-0 top-0 z-40 flex h-16 shrink-0 items-center gap-x-4 px-4 lg:hidden lg:px-6">
```

Comment:
What about having a sidebar component? As we have one for the desktop sidebar

Reply:
You mean one component for mobile and one for desktop?

Reply:
Yes we have one component for desktop "NavigationSideBar" and we could create a "MobileSideBar". We would simply conditionally render one in here

Reply:
I can do it in a separate PR

Reply:
All yours base on the implementation of your hide sidebar 🙃.